### DISCUSSION TITLE
Issue: how to let LangSmith only trace certain calls (toggle on / off for specific function calls)

### DISCUSSION THREAD

**Discussion by 3kywa1ke2** [langchain-ai/langsmith-sdk#834] [Q&A]
### Issue you'd like to raise.

Hi,

I saw the documents that says using the traceable decorator can toggle on and off the tracing for functions.

When I testing my application, it still traces all function calls regardless whether the decorator is added or not.

Is there a way that can make langsmith only trace a certain function, instead of tracing all functions?

Thank you!

**Comment by hinthornw** [comment]
You can toggle whole traces like this:

```
with langsmith.tracing_context(enabled={True|False}):
     ...
````

disabling a specific run within a larger trace isn't well-supported though

**Comment by hinthornw** [comment]
Feel free to re-open if this doesn't answer your question
